HCollections/HListT.fs

```diff
@@ -9,11 +9,17 @@ type HListT<'ts, 'elem> =
     | Empty of 'ts HList * Teq<'ts, unit>
     | Cons of 'ts HList * 'elem list * HListTConsCrate<'ts, 'elem>
 
-and private HListTConsEvaluator<'ts, 'elem, 'ret> =
+and private HListTConsEval<'ts, 'elem, 'ret> =
```

Comment:
Name clash :(. Thought it was better for the public type to have the full name

Reply:
Honestly I think you could just make the private type public.

Reply:
For HListT I'm not sure if that's a good idea. It'd require the end user to cast the stored HlistT and then pull the required elements themselves

Reply:
The internals are hidden, so it's not really a coupling. If in the future the things stored on HList is changed that new type can be private, and the ConsCrate just left as is. No major version bump.